openwisp_controller/connection/tests/test_api.py

```diff
@@ -175,6 +176,28 @@ def test_command_attributes(self, payload):
             self.assertEqual(response.status_code, 201)
             test_command_attributes(self, payload)
 
+    # for ensuring that only related connections are shown
+    def test_available_connections(self):
+        url = self._get_path('device_command_list', self.device_id)
+        response = self.client.get(url)
+        serializer = CommandSerializer(
+            instance=self.device_conn.device,
+            context={'device_id': self.device_id, 'request': response.wsgi_request},
+        )
+        device = self._create_device(
+            name='default.test.device2', mac_address='12:23:34:45:56:67'
+        )
+        self._create_config(device=device)
+        credentials_2 = self._create_credentials(name='Test Credentials 2')
+        device_conn2 = self._create_device_connection(
+            device=device, credentials=credentials_2
+        )
+        connections = serializer.fields['connection']
+        queryset = list(connections.get_queryset().values_list('id', flat=True))
+        queryset = [str(i) for i in queryset]
+        self.assertIn(str(self.device_conn.id), queryset)
+        self.assertNotIn(device_conn2.id, queryset)
+
```

Comment:
We generally test the presence/absence of desired elements in browsable API like this 

https://github.com/openwisp/openwisp-controller/blob/523f4dab1b84d4b0611085ec1fab60aeb65e7470/openwisp_controller/config/tests/test_api.py#L767-L775

Reply:
Thanks! Was looking for something like this